src/main/java/Parser.java

```diff
@@ -0,0 +1,170 @@
+     * required action.
+     *
+     * @param command command entered by the user.
+     * @param tasks current list of tasks
+     * @param ui current user interface instance
+     * @throws BlitzException if the command description is missing, or
+     *     if a request to print an empty list is made or if the user enters
+     *     an invalid command.
+     */
+    public static void parse(String command, TaskList tasks, Ui ui) throws BlitzException{
+        String[] keywords = command.split(" ");
+
+        //stores the first word (keyword) in the user input
+        String firstKeyword = keywords[0];
+
+        //stores whether keyword is a task or calling a feature
+        boolean isTask = firstKeyword.equals("todo") || firstKeyword.equals("event")
+                || firstKeyword.equals("deadline");
+        boolean isFeature = firstKeyword.equals("list") || firstKeyword.equals("done")
+                || firstKeyword.equals("delete") || firstKeyword.equals("find");
+
+        //suppose it's a valid keyword
+        if(isTask || isFeature) {
+
+            //when keyword is not followed by anything
+            if (isTask && command.length() < firstKeyword.length() + 2) {
+                throw new BlitzException("The description of a " + firstKeyword
+                        + " cannot be empty.");
+            }
+
+            if (isFeature) {
+                switch (firstKeyword) {
+                case "list":
+                    if (tasks.size() == 0) {
+                        throw new BlitzException("No items added to list yet!");
+                    }
+                    ui.printLine();
+                    ui.printList(tasks, "Here are the tasks in your list:");
+                    ui.printLine();
+                    break;
+                case "done":
+                    try {
+                        int index = getIndex(command, tasks.size());
+                        Task finished = tasks.get(index);
+                        finished.markAsDone();
+                        ui.printFormatted("Nice! I've marked this task as done:\n" + "\t\t"
+                                + finished);
+                    } catch (BlitzException ex) {
+                        ui.printFormatted(ex.toString());
+                    }
+                    break;
+                case "delete":
+                    try {
+                        int index = Parser.getIndex(command, tasks.size());
+                        Task deleted = tasks.deleteTask(index);
+                        ui.printFormatted("Noted. I've removed this task:" + "\n\t\t" + deleted
+                                + "\n\tNow you have " + tasks.size() + " tasks in the list.");
+                    } catch (BlitzException ex) {
+                        ui.printFormatted(ex.toString());
+                    }
+                    break;
+                default:
+                    String findKeyword = keywords[1];
+
+                    if(tasks.size() == 0) {
```

Comment:
Just a small coding standard violation: `if` statements should be followed with a whitespace.